src/CommonDBRelation.php

```diff
@@ -2145,4 +2146,30 @@ public static function getMemberPosition(string $class): int
             return 0;
         }
     }
+
+    public function check($ID, int $right, ?array &$input = null): void
+    {
+        $error_types = [];
+        if (static::$itemtype_1 !== 'itemtype' && static::$mustBeAttached_1 === true && (!isset($input[static::$items_id_1]) || empty($input[static::$items_id_1]))) {
```

Comment:
When `static::$itemtype_1` is `itemtype`, the `$input[static::$items_id_1]` value is required. Also, in this case, `$input[static::$itemtype_1]` is required too.

For `static::$itemtype_1= Computer::class; static::$items_id_1 = 'computers_id';`, the input must be something like `['computer_id' => 123456]`, but for `static::$itemtype_1= 'itemtype'; static::$items_id_1 = 'items_id';`, the input must be something like `['itemtype' => Computer::class, 'items_id' => 123456]`.

Please note that `static::$itemtype_1`  value may have a suffix, for instance `itemtype_peripheral`.

Reply:
OK, I'll change that (and I'll add some tests)

Reply:
Done in commit c666af5086eea2abd66c45bf0f18f9424ec0ce15